File: app/src/main/java/com/example/guest/newyorktimesclient/di/components/ApplicationComponent.java
package com.example.guest.newyorktimesclient.di.components;

import android.content.Context;

import com.example.guest.newyorktimesclient.di.modules.ApplicationModule;

import javax.inject.Singleton;

import dagger.Component;
import retrofit2.Retrofit;

/**
 * Created by l1maginaire on 2/28/18.
 */

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {
    Retrofit exposeRetrofit();
    Context exposeContext();
}

File: app/src/main/java/com/example/guest/newyorktimesclient/di/components/ApplicationComponent.kt
package com.example.guest.newyorktimesclient.di.components

import com.example.guest.newyorktimesclient.NewsApp
import com.example.guest.newyorktimesclient.di.modules.*

import javax.inject.Singleton

import dagger.Component

/**
 * Created by l1maginaire on 2/28/18.
 */

@Singleton
@Component(modules = [
    ApplicationModule::class,
    SchedulerProviderModule::class,
    GsonModule::class,
    NetworkModule::class,
    ApiClientModule::class,
    ImageLoaderModule::class
/*,
    DispatchersProviderModule::class,
    DatabaseModule::class,
    UseCaseProviderModule::class,
    UtilsModule::class*/
])
interface ApplicationComponent {
    fun inject(application: NewsApp)
//    fun inject(pushNotificationReceiverService: PushNotificationReceiverService)
//
//    fun plus(takePhotoActivityModule: TakePhotoActivityModule): TakePhotoActivityComponent
//    fun plus(viewTakenPhotoActivityModule: ViewTakenPhotoActivityModule): ViewTakenPhotoActivityComponent
//    fun plus(photosActivityModule: PhotosActivityModule): PhotosActivityComponent
//    fun plus(uploadPhotoServiceModule: UploadPhotoServiceModule): UploadPhotoServiceComponent
//    fun plus(receivePhotosServiceModule: ReceivePhotosServiceModule): ReceivePhotosServiceComponent
//    fun plus(settingsActivityModule: SettingsActivityModule): SettingsActivityComponent
}